scaleway/resource_k8s_cluster_beta.go

```diff
@@ -418,56 +430,66 @@ func resourceScalewayK8SClusterBetaCreate(d *schema.ResourceData, m interface{})
-		defaultPoolReq.PlacementGroupID = scw.StringPtr(expandID(placementGroupID.(string)))
-	}
+	if _, ok := d.GetOk("default_pool"); ok {
+		defaultPoolReq := &k8s.CreateClusterRequestPoolConfig{
+			Name:        "default",
+			NodeType:    d.Get("default_pool.0.node_type").(string),
+			Autoscaling: d.Get("default_pool.0.autoscaling").(bool),
+			Autohealing: d.Get("default_pool.0.autohealing").(bool),
+			Size:        uint32(d.Get("default_pool.0.size").(int)),
+			Tags:        expandStrings(d.Get("default_pool.0.tags")),
+		}
 
-	defaultPoolReq.MinSize = scw.Uint32Ptr(uint32(d.Get("default_pool.0.min_size").(int)))
+		if placementGroupID, ok := d.GetOk("default_pool.0.placement_group_id"); ok {
+			defaultPoolReq.PlacementGroupID = scw.StringPtr(expandID(placementGroupID.(string)))
+		}
 
-	if maxSize, ok := d.GetOk("default_pool.0.max_size"); ok {
-		defaultPoolReq.MaxSize = scw.Uint32Ptr(uint32(maxSize.(int)))
-	} else {
-		defaultPoolReq.MaxSize = scw.Uint32Ptr(defaultPoolReq.Size)
-	}
+		defaultPoolReq.MinSize = scw.Uint32Ptr(uint32(d.Get("default_pool.0.min_size").(int)))
 
-	if containerRuntime, ok := d.GetOk("default_pool.0.container_runtime"); ok {
-		defaultPoolReq.ContainerRuntime = k8s.Runtime(containerRuntime.(string))
-	}
+		if maxSize, ok := d.GetOk("default_pool.0.max_size"); ok {
+			defaultPoolReq.MaxSize = scw.Uint32Ptr(uint32(maxSize.(int)))
+		} else {
+			defaultPoolReq.MaxSize = scw.Uint32Ptr(defaultPoolReq.Size)
+		}
 
-	req.DefaultPoolConfig = defaultPoolReq
+		if containerRuntime, ok := d.GetOk("default_pool.0.container_runtime"); ok {
+			defaultPoolReq.ContainerRuntime = k8s.Runtime(containerRuntime.(string))
+		}
 
-	res, err := k8sAPI.CreateCluster(req)
-	if err != nil {
-		return err
+		req.Pools = []*k8s.CreateClusterRequestPoolConfig{defaultPoolReq}
 	}
 
-	d.SetId(newRegionalId(region, res.ID))
-
-	err = waitK8SClusterReady(k8sAPI, region, res.ID) // wait for the cluster status to be ready
+	res, err := k8sAPI.CreateCluster(req)
 	if err != nil {
 		return err
 	}
 
-	if d.Get("default_pool.0.wait_for_pool_ready").(bool) { // wait for the pool status to be ready (if specified)
-		pool, err := readDefaultPool(d, m) // ensure that 'default_pool.0.pool_id' is set
+	if _, ok := d.GetOk("default_pool"); ok {
+		err = waitK8SClusterReady(k8sAPI, region, res.ID)
 		if err != nil {
 			return err
 		}
+		if d.Get("default_pool.0.wait_for_pool_ready").(bool) { // wait for the pool status to be ready (if specified)
+			pool, err := readDefaultPool(d, m) // ensure that 'default_pool.0.pool_id' is set
+			if err != nil {
+				return err
+			}
+
+			err = waitK8SPoolReady(k8sAPI, region, expandID(pool.ID))
+			if err != nil {
+				return err
+			}
+		}
 
-		err = waitK8SPoolReady(k8sAPI, region, expandID(pool.ID))
+	} else {
+		err = waitK8SClusterPoolRequired(k8sAPI, region, res.ID)
 		if err != nil {
 			return err
 		}
 	}
 
+	d.SetId(newRegionalId(region, res.ID))
```

Comment:
Why do you do that here? This is done in the Read

Reply:
:thinking: This needs to be done here or else the Read will have no ID to read from :sweat_smile: